lib/provable-generic.ts

```diff
@@ -173,6 +173,21 @@ function createDerivers<Field>(): {
       if (!complexTypes.has(typeof typeObj))
         throw Error(`provable: unsupported type "${typeObj}"`);
 
+      if (display(typeObj) === 'Struct')
```

Comment:
* I think this should be moved below the `if (isProvable(...)) { ... }` branch so we don't prevent valid provable types called `Struct`
* To further harden what types we accept, I think we should add another check with generalizes this one: If, after the `isProvable()` case, we have `typeof typeObj === "function"`, then this is almost definitely a mistake and we should throw an error (a more generic error than this one of course, so keep this error for the specific case of Struct). the final case where we recursive into the object properties should only be done for anonymous objects.

Reply:
Added the extra check! If the error message isn't good enough, happy to change it!

Reply:
Thanks!